didman/didman.go

```diff
@@ -220,6 +223,77 @@ func (d *didman) GetContactInformation(id did.DID) (*ContactInformation, error)
 	return nil, nil
 }
 
+func (d *didman) SearchOrganizations(query string, didServiceType *string) ([]OrganizationSearchResult, error) {
```

Comment:
method is rather lengthy, maybe extract the two filtering parts into their own funcs

Reply:
I'll refactor it further.